src/prefect/environments/storage/base.py

```diff
@@ -38,10 +42,13 @@ def __init__(
         labels: List[str] = None,
         add_default_labels: bool = None,
         stored_as_script: bool = False,
+        script_path: str = None,
```

Comment:
Since this doesn't make sense for all storage types, perhaps this shouldn't be added to the base class? AFAICT it only makes sense for `S3` and `GCS` storage (currently).

Reply:
Also wondering if `local_script_path` would be help disambiguate from a path in remote storage? No strong thoughts.

Reply:
Good call yeah I'll move it to those classes only and change the name. Was originally thinking about doing something like that instead